pegasus/src/social_metadata.rb

```diff
@@ -0,0 +1,155 @@
+        video: {youtube_key: "QvyTEx1wyOY", width: 640, height: 360}
+      },
+      "default" => {
+        title: I18n.t(:csedweek_og_title),
+        description: I18n.t(:csedweek_og_description),
+        image: {path: "/images/cs-is-everything-thumbnail.png", width: 1200, height: 627},
+        video: {youtube_key: "QvyTEx1wyOY", width: 640, height: 360}
+      }
+    },
+    "hourofcode.com" => {
+      "soon-hoc" => {
+        title: I18n.t(:og_title_soon),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "week-before-hoc" => {
+        title: I18n.t(:og_title_soon),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "actual-hoc" => {
+        title: I18n.t(:og_title_here),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "default" => {
+        title: hoc_s(:meta_tag_og_title),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      }
+    },
+    "challenge" => {
+      "soon-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1920/social-media/hoc-student-challenge.jpg", width: 1920, height: 1080}
+      },
+      "week-before-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1920/social-media/hoc-student-challenge.jpg", width: 1920, height: 1080}
+      },
+      "actual-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1220/social-media/celeb-challenge.jpg", width: 1220, height: 640}
+      },
+      "default" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1220/social-media/celeb-challenge.jpg", width: 1220, height: 640}
+      }
+    }
+  }
```

Comment:
So, right now all of the versions for each page are the same for each hoc_mode. You're just setting up the system where they could all be different and then we'll customize next year?

Reply:
The image actually changes for the second two, but you're right that there's some duplication too.  At this point, any duplication gives us a framework for future customisation.